src/test/unit/scheduler_unittest.cc

```diff
@@ -51,10 +51,15 @@ extern "C" {
     bool taskPidRan = false;
     bool taskFilterReady = false;
     bool taskPidReady = false;
+    uint8_t activePidLoopDenom = 1;
 
     // set up micros() to simulate time
     uint32_t simulatedTime = 0;
     uint32_t micros(void) { return simulatedTime; }
+    int32_t clockCyclesToMicros(int32_t x) { return x/10;}
+    int32_t clockCyclesTo10thMicros(int32_t x) { return x;}
+    int32_t clockMicrosToCycles(int32_t x) { return x*10;}
+    int32_t getCycleCounter(void) {return simulatedTime * 10;}
```

Comment:
should be `uint32_t`.  Also ideally there should be a unit test that handles the `getCycleCounter` integer overflow so we can be sure it's handled correctly.

Reply:
For integer comparison it is always best to use signed. Consider a simple 8 bit counter `c` . Pick any value, add say 10 to give a terminal count, `t`.